### ISSUE TITLE
Offsets should be in bytes for HTML clipboard format

### ISSUE THREAD

**Issue by taku0** [jaraco/jaraco.windows#24]
`StartFragment`, `EndFragment`, and other offsets of [HTML clipboard format](https://learn.microsoft.com/en-us/windows/win32/dataxchg/html-clipboard-format) is defined as "in bytes" but [`HTMLSnipped`](https://github.com/jaraco/jaraco.windows/blob/e931a4b484850e48bc4804b06060b24d78bfebd3/jaraco/windows/clipboard.py#L85-L95) class interprets them as character offsets.

This results in invalid HTML fragments if the HTML contains multi-byte characters.  Steps to reproduce:

1. Copy some [emojis](https://emojipedia.org/smileys) from browser.
2. Evaluate the following snippet in Python REPL:
   ```python
   import jaraco.windows.clipboard as wclip

   wclip.get_html().fragment
   ```

Expected result: valid HTML fragment without `<!--EndFragment -->` comment.
Actual result: invalid HTML fragment including `<!--EndFragment -->` and `</body></html>`, depending on number of characters copied.

`set_html` should also be fixed.

**Comment by jaraco** [comment]
Thanks for the report!

Would you be interested in working on a fix?

**Comment by taku0** [comment]
Yes, I'll work on it over the weekend.

**Comment by jaraco** [comment]
Fix released in 5.9.1
